test/puppetlabs/puppetdb/config_test.clj

```diff
@@ -186,7 +186,32 @@
                                           write-database-config-out)
                        configure-read-db)
             thirty-seconds-in-millis 30000]
-        (is (= (get-in config [:database :schema-check-interval]) thirty-seconds-in-millis))))))
+        (is (= (get-in config [:database :schema-check-interval]) thirty-seconds-in-millis))))
+
+    (let [no-migrator {:database {:classname "something"
+                                  :subname "stuff"
+                                  :subprotocol "more stuff"
+                                  :username "someone"
+                                  :password "something"}}
+          migrator (update no-migrator :database assoc
+                           :migrator-username "admin"
+                           :migrator-password "admin")]
+
+      (testing "migrator-username"
+        (let [config (configure-db no-migrator)]
+          (is (= "someone" (get-in config [:database :migrator-username])))
+          (is (= "someone" (get-in config [:read-database :migrator-username]))))
```

Comment:
I wonder about including the migrator-username/password in the read-database section of the config. I'm not sure how we handle DB migrations if someone is running a hot standby using the [read-database](https://puppet.com/docs/puppetdb/latest/configure.html#read-database-settings) config section. Do we know if migrations are performed on the hot standby in this setup? If they're not we might not want to allow the migrator user to show up in this config section.

Reply:
Hmm, no strong opinion.  I'd just kept treating the read and write sections mostly the same, asn I think we had been, and fwiw, just having the migrator info in the read db shouldn't actually cause anything to happen.  The actual migration should still be guarded by `[database] migrate`.

Reply:
But I'm happy to see about adjusting it to elide the migrator settings altogether if we think that's preferable.